client/app/libs/accounts.coffee

```diff
@@ -84,28 +214,48 @@ module.exports =
         return state
 
 
-    # Take a given state:
-    # - removes unwanted keys for the final config (state-only concerned)
-    # - bind desired keys (about security concerns) to the given state (here, it
-    # translates from the <select> choice to DS Model's boolean fields).
-    # - extract name and label from the mail login field (aka email identifier)
-    sanitizeConfig: (state) ->
-        excludes = [
-            'OAuth'
-            'alert'
-            'isBusy'
-            'isDiscoverable'
-            'success'
-            'enableSubmit'
-            'imapSecurity'
-            'smtpSecurity'
-        ]
-        [name, ...] = state.login.split '@'
-
-        _.extend _.omit(state, excludes),
-            label:   state.login
-            name:    name
-            imapSSL: state.imapSecurity is 'ssl'
-            imapTLS: state.imapSecurity is 'starttls'
-            smtpSSL: state.smtpSecurity is 'ssl'
-            smtpTLS: state.smtpSecurity is 'starttls'
+    # @context : 'imap' or 'smtp'
+    # Return unprefixed props
+    # to the Server component
+    filterPropsByProvider: (props={}, protocol) ->
```

Comment:
I think this whole thing try to be too smart with managing different protocols, let's instead pass 
protocol & the whole account object to the Server component, have the Server component handle concatenatenating protocol and name, and have slugs / id / name be the same camelCase Value.
Form Input  can then just pass their name as second parameter to the onChange listener.